controller_manager:
  ros__parameters:
    update_rate: 400 # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_state_broadcaster:
  ros__parameters:
    publish_rate: 50.0

joint_trajectory_controller:
    ros__parameters:
      interface_name: position

      joints:
        - front_left_first_joint
        - front_left_second_joint
        - front_left_third_joint

        - front_right_first_joint
        - front_right_second_joint
        - front_right_third_joint

        - rear_left_first_joint
        - rear_left_second_joint
        - rear_left_third_joint

        - rear_right_first_joint
        - rear_right_second_joint
        - rear_right_third_joint

      command_interfaces:
        # - velocity
        - position
      state_interfaces:
        - position
        - velocity
        # - acceleration

      state_publish_rate: 100.0
      action_monitor_rate: 20.0
      # gains.front_left_knee_joint: 1.0
